### PR TITLE
Add user file open preference in context & send user's file open pref while opening files

### PR DESCRIPTION
**Pull request by vivek3003**
Apps can read where user's prefer to open files. The current values are:
Teams, Desktop (Desktop app of the file such as excel/powerpoint), Web (browser)

### CODE REVIEW

**Comment by ydogandjiev** [comment]
@vivek3003, were you planning to also add a query API for the setting that apps can use to figure out what the current setting is?

**Comment by ydogandjiev** [comment]
> @vivek3003, were you planning to also add a query API for the setting that apps can use to figure out what the current setting is?

File: src/public/interfaces.ts [hunk 1]
```diff
@@ -394,6 +403,11 @@ export interface Context {
    * Time when the user clicked on the tab
    */
   userClickTime?: number;
+
+  /**
+   * Where the user prefers the file to be opened from by default during file open
+   */
+  userFileOpenPreference?: FileOpenPreference;
```
Comment (ydogandjiev) [inline src/public/interfaces.ts:410]: Where is this information stored on the client? Does it ever require making a network call or is the information always available locally?

File: src/public/interfaces.ts [hunk 2]
```diff
@@ -158,6 +158,15 @@ export interface LocaleInfo {
   longTime: string;
 }
 
+/**
+ * Allowed user file open preferences
+ */
+export enum FileOpenPreference {
+  Teams = 'teams',
```
Comment (ydogandjiev) [inline src/public/interfaces.ts:165]: Any chance we can make this a little more generic - e.g. Hub, InApp, or something like that? We're soon going to be replacing the Teams Client SDK with the MetaOS App SDK which will used to host apps in a number of different Office hubs like Outlook, Office.com, etc.

**Review by ydogandjiev — COMMENTED** [review]
As discussed offline, lets please review these changes with @sachingupta and myself before committing this PR.

File: src/public/interfaces.ts (hunk 1 above)
Comment (vivek3003) [reply src/public/interfaces.ts:410]: This is stored and read from the preferences store. It's read synchronously while passing to the context. 
No network called involved.

File: src/public/interfaces.ts (hunk 2 above)
Comment (vivek3003) [reply src/public/interfaces.ts:165]: Yes, we should be able to. But we will need to write a mapping function in teams to read&pass this values as inApp.

Comment (vivek3003) [reply src/public/interfaces.ts]: Have updated to 'inApp'

**Review by ydogandjiev — APPROVED** [review]

**Pull request merged**